fvm/evm/handler/addressAllocator.go

```diff
@@ -53,3 +61,24 @@ func (aa *AddressAllocator) AllocateAddress() (types.Address, error) {
 
 	return target, nil
 }
+
+func MakeCOAAddress(index uint64) types.Address {
+	var addr types.Address
+	prefixIndex := types.AddressLength - uint64ByteSize
+	copy(addr[:prefixIndex], FlowEVMCOAAddressPrefix[:])
+	binary.BigEndian.PutUint64(addr[prefixIndex:], index)
+	return addr
```

Comment:
maybe extract both of `MakeCOAAddress` and `MakePrecompileAddress` into a helper method like `makePrefixedAddress(index uint64, prefix [12]byte)`

Reply:
fixed at [a6abd9b](https://github.com/onflow/flow-go/pull/5233/commits/a6abd9b9ae7efe8c96f8b3ae11a11004bfefa3c7)